### ISSUE TITLE
Regression when using a custom compiler in version 1.4.0

### ISSUE THREAD

**Issue by Davian99** [peerigon/angular-expressions#58]
Hi, this is just a heads up for everyone that uses this library with a custom implementation of `compile`. I know this is not the case for the majority, but I spent way more time that necessary debugging this.

So, we use the `Lexer` and the `Parser` for, duh, parsing expressions, but we handle them with a custom implementation of `compile`. According to the `1.4.0` release, `handleThis` is set to true (the previous behaviour) by default, but only when using the default `compile` function, not when creating a `Parser` manually.

In this case, the default behaviour is actually NOT handling this a context, but as a variable. The fix also is super simple, just add `handlethis: true` as an option to the `Parser` and everything works as before.

I understand that our usecase is not in the majority, but maybe we can have this option set by default to true in the `Parser`?

**Comment by edi9999** [comment]
I see, it makes sense.

Would the following diff work for you ? 

```diff
diff --git a/lib/parse.js b/lib/parse.js
index 8366311..18220c7 100644
--- a/lib/parse.js
+++ b/lib/parse.js
@@ -2785,6 +2785,8 @@ ASTInterpreter.prototype = {
 var Parser = function Parser(lexer, $filter, options) {
 	this.lexer = lexer;
 	this.$filter = $filter;
+	options = options || {};
+	options.handleThis = options.handleThis != null ? options.handleThis : true;
 	this.options = options;
 	this.ast = new AST(lexer, options);
 	this.ast.selfReferential = {
```

**Comment by edi9999** [comment]
Technically, I think that by adding handleThis that defaulted to false be basically broke backwwards compatibility, that's why you found this issue right ??

**Comment by edi9999** [comment]
I've just fixed this in v1.4.2

**Comment by Davian99** [comment]
> Technically, I think that by adding handleThis that defaulted to false meant that we broke backwwards compatibility (for users of Lexer and Parser), that's why you found this issue right ??

Yes, exactly how I found it.

Thanks so much for looking into and fixing it so fast.
